### ISSUE TITLE
[Q]: Wandb charts show time (minutes), but I want seconds.

### ISSUE THREAD

**Issue by kailashg26** [wandb/wandb#9387]
### Ask your question

I’m working with Weights and Biases and previously, the x-axis time was shown in seconds. However, now all my runs are showing the x-axis in minutes instead of seconds, and I need to change it back to seconds.

Currently using firefox browser.

**Comment by aajais** [comment]
Hi @kailashg26, Thanks for reaching out! Could you share more details about the issue you’re experiencing? Are you referring to the x-axis on the System Metrics charts? If possible, please provide a link to the chart where you’re seeing this behavior, along with your wandb SDK version. Thanks.

**Comment by kailashg26** [comment]
Hi @aajais, here's the screenshot.[

![Image](https://github.com/user-attachments/assets/68b15cd8-4d89-4944-af79-135fc5760d91)

](url)


Previously it used be in seconds. Somehow it now shows in minutes.

**Comment by aajais** [comment]
For these charts, if you’re using Relative Time, the scaling adjusts automatically based on how long your script runs. If it exceeds a few hours, the time unit will switch to hours. If you’d like, I’d be happy to submit a feature request for more control over this behavior.

**Comment by kailashg26** [comment]
Yes, please submit a request. That would be helpful.

**Comment by aajais** [comment]
The timeline for this isn’t fully clear at the moment. I’ll check in with our engineering team and follow up once I have more details.

**Comment by kailashg26** [comment]
Or can we somewhere in the code?

**Comment by kailashg26** [comment]
Hi @aajais , my SDK is 0.18.3
Do you have any update regarding this?

**Comment by aajais** [comment]
Hi @kailashg26, Still no updates on this, it might take a little while to get processed. In the meantime, as a workaround, you can manually log the second value and use it as the x-axis for your plots. You could do something like this: Let me know if this works!
```
start_time = time.time()

# training_loop
    elapsed_seconds = time.time() - start_time
    wandb.log({"elapsed_seconds": elapsed_seconds})
```

**Comment by kailashg26** [comment]
Hi @aajais,

                        if self._device.type == "cuda" and self._log_peak_memory_stats:
                            log_dict.update(training.get_memory_stats(device=self._device))
                        if self._clip_grad_norm is not None:
                            log_dict.update({"grad_norm": grad_norm})
                       
                        self._metric_logger.log_dict(log_dict, step=self.global_step)
                   
                    running_loss = 0
                    num_tokens = 0
                    t0 = time.perf_counter()
               
                if (
                    curr_epoch == 0
                    and self.profiler_profile_memory
                    and idx == self.profiler_wait_steps + self.profiler_warmup_steps + self.profiler_active_steps
                ):
                    torch.cuda.memory._record_memory_history(enabled=None)
               
                prof.step()

            self.epochs_run += 1
            start_save_checkpoint = time.perf_counter()
            log.info("Starting checkpoint save...")
            self.save_checkpoint(epoch=curr_epoch)
            log.info(
                "Checkpoint saved in {:.2f} seconds.".format(
                    time.perf_counter() - start_save_checkpoint
                )
            )

**Comment by aajais** [comment]
Hi @kailashg26, Did you happen to change the x-axis metric for the chart in the UI to time_seconds metric? If not, could you give that a try? Also, can you share a link to your workspace where you are seeing this issue?

**Comment by kailashg26** [comment]
https://wandb.ai/kg26/torchtune/runs/k3ybxj73?nw=nwuserkg26

It's kinda showing in all the charts. There's no option for time_seconds metric.

**Comment by aajais** [comment]
It looks like the [metric](https://wandb.ai/kg26/torchtune/runs/k3ybxj73/files/wandb-summary.json) wasn’t logged. Your code looks good, so could you try rerunning your script to see if it appears?

If the issue persists, could you share the debug logs with us? These logs will help us troubleshoot. You can find them in the directory: directory `./wandb/run-{date_time}-{run_id}/logs` on the machine where you ran your code. 

If you want to download the data, you can use the Import/Export API. Here’s the [documentation](https://docs.wandb.ai/guides/track/public-api-guide/#export-run-data) that walks you through the process.

**Comment by blutjens** [comment]
Is there an update on this? I am experiencing a related bug that seems like it would be fixed by manual control over the x-axis unit. The bug is that my wandb GUI switches back and forth between Time (min) and Time (seconds) every ~6 seconds. Furthermore, the data is incorrectly displayed whenever the GUI chooses to show Time (seconds) as the x-axis. Here are two screenshots of the visualizations that the GUI switches in between:

Version 1: Correct
<img width="1618" height="800" alt="Image" src="https://github.com/user-attachments/assets/5cca0ab6-5100-40d6-8d17-eb3c0644f880" />

Version 2: Incorrect
<img width="1616" height="812" alt="Image" src="https://github.com/user-attachments/assets/5bd2b85f-d8a3-4475-993f-b9df40ff46a1" />

**Comment by aajais** [comment]
@blutjens thanks for flagging this! We’ve pushed a fix for the issue, and it should now be resolved. Let me know if you’re still seeing any problems.

**Comment by blutjens** [comment]
@aajais Thank you! The issue persists in a test I just did in an incognito window, on a mac Tahoe 26.1 + chrome, with the same ongoing run, plotting "GPU 0 Utilization" over "Relative Time (Wall)". I did find a workaround by manually setting the x-axis unit via: "expressions", "x-axis expression", "_absolute_runtime / 60 / 60" and it doesn't refresh and change the x-axis anymore.

**Comment by manishm-cw** [comment]
@kailashg26 are you still experiencing this issue? Relative time should show in seconds or mins depending on how long your script runs.
